spec/helpers.rb

```diff
@@ -1,7 +1,7 @@
 module Helpers
   def login
     user = FactoryBot.create(:user)
-    post "/login", params: { "email" => user.email, "password" => user.password }
+    post "/login", params: { "email" => user.email, "password" => User::TEST_PASSWORD }
```

Comment:
Looks like you are defining a constant for `TEST_PASSWORD` so that you can use the same password in test and in factory. But you can also pass the password into your `FactoryBot#create` so that you can set it from the spec by using [transient](https://devhints.io/factory_bot).

Reply:
@naush I don't think I have gotten the full grasp of how transient will work in this context because from the documentation it says transient is an `after_create` command.

Reply:
@kenneth051 Ah OK maybe I was wrong about that. I was thinking something like this:

```rb
factory :user do
  transient do
    password 'changeme'
  end

  before :create do |user, options|
    user.password = BCrypt::Password.create(options.password) if options.password
  end
end

create(user, password: 'mypassword')
```

So this way in your test you could reuse the same password for when you call FactoryBot.create and controller parameter.

Reply:
Feel free to skip this feedback if it doesn't work/or apply!

Reply:
ok